tycho-bundles/org.eclipse.tycho.p2.resolver.impl/src/main/java/org/eclipse/tycho/p2/remote/RemoteRepositoryCacheManager.java

```diff
@@ -77,14 +77,19 @@ public File createCache(URI repositoryLocation, String prefix, IProgressMonitor
 
     @Override
     public File createCacheFromFile(URI remoteFile, IProgressMonitor monitor) throws ProvisionException, IOException {
+        File cacheFile = getCacheFile(remoteFile, getCacheDirectory());
         if (offline) {
-            File cacheFile = getCacheFile(remoteFile, getCacheDirectory());
             if (cacheFile != null) {
                 return cacheFile;
             }
-            throw new ProvisionException(getFailureStatus(remoteFile));
```

Comment:
Shouldn't the "throw" here be preserved in case of offline + cache miss? I have the impression modified code would try to download the file in that case.

Reply:
yes, not sure why i removed that, force pushed a new change